### ISSUE TITLE
[Bug] [V4] Logging in

### ISSUE THREAD

**Issue by Revadike** [DoctorMcKay/node-steam-user#246]
Sometimes, when I try to log into steam it needs several tries. I think it times out after 2 minutes. Thus, it takes several minutes to log in. While other times, I can just log in instantly. No code has changed. V3 didn't have this problem. Maybe this problem was already known.  I could provide additional debugging information, if needed. Here are my login options:
```

    config.credentials.steam.twoFactorCode = steam.auth.getAuthCode(config.credentials.steammobile.shared_secret);
    config.credentials.steam.logonID = Date.now();
    config.credentials.steam.rememberPassword = true;
    config.credentials.steam.dontRememberMachine = false;
    config.credentials.steam.autoRelogin = true;
    config.credentials.steam.machineName = `BOT@${config.servermode ? `SERVER` : `CLIENT`}`;

    steam.user.logOn(config.credentials.steam);
```

**Comment by DoctorMcKay** [comment]
For starters, logonID is a 32-bit int so using Date.now() isn't going to work how you expect.

Please add `steam.user.on('debug', msg => console.log('[steam-user debug] ' + msg));` so we can get some debugging output to work with.

**Comment by Revadike** [comment]
Yes, I did that before. I observed the behavior I described before. But here is the debugger log:
```
 2019-04-04 | 00:18:22.911 | INFO  | initBot                        | Initializing bot by Revadike
 2019-04-04 | 00:18:22.920 | INFO  | determineBotMode               | Bot is operating in CLIENT mode
 2019-04-04 | 00:18:23.183 | DEBUG | emit                           | Using WebSocket; we rolled 49 and percent to use WS is 50
 2019-04-04 | 00:18:23.972 | DEBUG | emit                           | WebSocket connection success; now logging in
 2019-04-04 | 00:18:23.974 | DEBUG | emit                           | Sending message: ClientLogon
 2019-04-04 | 00:18:24.527 | DEBUG | emit                           | Unhandled message: ClientServersAvailable
 2019-04-04 | 00:18:24.528 | DEBUG | emit                           | Unhandled message: ClientServersAvailable
 2019-04-04 | 00:18:24.530 | DEBUG | emit                           | Handled message: ClientLogOnResponse
 2019-04-04 | 00:18:24.536 | DEBUG | emit                           | Sending message: ClientRequestFriendData
 2019-04-04 | 00:18:24.539 | DEBUG | emit                           | Sending message: ClientPICSChangesSinceRequest
 2019-04-04 | 00:18:24.543 | DEBUG | emit                           | Sending message: ClientRequestItemAnnouncements
 2019-04-04 | 00:18:24.545 | DEBUG | emit                           | Sending message: ClientRequestCommentNotifications 2019-04-04 | 00:18:24.548 | DEBUG | emit                           | Sending message: ClientChatRequestOfflineMessageCount
 2019-04-04 | 00:18:24.553 | DEBUG | emit                           | Handled message: ClientAccountInfo
 2019-04-04 | 00:18:24.554 | DEBUG | emit                           | Handled message: ClientAccountInfo
 2019-04-04 | 00:18:24.555 | DEBUG | emit                           | Handled message: ClientEmailAddrInfo
 2019-04-04 | 00:18:24.586 | DEBUG | emit                           | Handled message: ClientFriendsList
 2019-04-04 | 00:18:24.741 | DEBUG | emit                           | Sending message: ClientRequestFriendData
 2019-04-04 | 00:18:24.744 | DEBUG | emit                           | Handled message: ClientPlayerNicknameList
 2019-04-04 | 00:18:24.746 | DEBUG | emit                           | Handled message: ClientFriendsGroupsList
 2019-04-04 | 00:18:24.779 | DEBUG | emit                           | Unhandled message: ClientServersAvailable
 2019-04-04 | 00:18:24.971 | DEBUG | emit                           | Handled message: ClientPICSChangesSinceResponse
 2019-04-04 | 00:18:24.972 | DEBUG | emit                           | Handled message: ClientCommentNotifications
 2019-04-04 | 00:18:25.017 | DEBUG | emit                           | Handled message: ClientChatOfflineMessageNotification
 2019-04-04 | 00:18:25.038 | DEBUG | emit                           | API POST request to https://api.steampowered.com/ISteamUserAuth/AuthenticateUser/v0001/: 200
 2019-04-04 | 00:18:25.044 | DEBUG | emit                           | Sending message: ClientRequestWebAPIAuthenticateUserNonce
 2019-04-04 | 00:18:25.474 | DEBUG | emit                           | Sending message: ClientPICSChangesSinceRequest
 2019-04-04 | 00:18:55.983 | DEBUG | emit                           | Sending message: ClientPICSChangesSinceRequest
 2019-04-04 | 00:19:26.484 | DEBUG | emit                           | Sending message: ClientPICSChangesSinceRequest
 2019-04-04 | 00:19:56.989 | DEBUG | emit                           | Sending message: ClientPICSChangesSinceRequest
 2019-04-04 | 00:20:23.988 | DEBUG | emit                           | WebSocket disconnected with error: Ping timeout
 2019-04-04 | 00:20:24.991 | DEBUG | emit                           | Using WebSocket; we rolled 20 and percent to use WS is 50
 2019-04-04 | 00:20:25.077 | DEBUG | emit                           | WebSocket connection success; now logging in
 2019-04-04 | 00:20:25.079 | DEBUG | emit                           | Sending message: ClientLogon
 2019-04-04 | 00:20:25.576 | DEBUG | emit                           | Handled message: ClientLogOnResponse
 2019-04-04 | 00:20:25.578 | DEBUG | emit                           | Using TCP; we rolled 63 and percent to use WS is 50
 2019-04-04 | 00:20:25.579 | DEBUG | emit                           | Connecting to TCP CM: 162.254.196.84:27017
 2019-04-04 | 00:20:25.599 | DEBUG | emit                           | TCP connection established
 2019-04-04 | 00:20:25.616 | DEBUG | emit                           | Handled message: ChannelEncryptRequest
 2019-04-04 | 00:20:25.617 | DEBUG | emit                           | Channel encrypt request: protocol 1, universe 1, nonce 7b9b265703c32a0f3e9ec52db5257436, 0 remaining bytes
 2019-04-04 | 00:20:25.618 | DEBUG | emit                           | Sending message: ChannelEncryptResponse
 2019-04-04 | 00:20:25.642 | DEBUG | emit                           | Handled message: ChannelEncryptResult
 2019-04-04 | 00:20:25.643 | DEBUG | emit                           | Sending message: ClientLogon
 2019-04-04 | 00:20:26.152 | DEBUG | emit                           | Got incomplete message; expecting 7408 more bytes
 2019-04-04 | 00:20:26.153 | DEBUG | emit                           | Got incomplete message; expecting 7408 more bytes
 2019-04-04 | 00:20:26.171 | DEBUG | emit                           | Unhandled message: ClientServersAvailable
 2019-04-04 | 00:20:26.172 | DEBUG | emit                           | Unhandled message: ClientServersAvailable
 2019-04-04 | 00:20:26.172 | DEBUG | emit                           | Handled message: ClientLogOnResponse
 2019-04-04 | 00:20:26.175 | INFO  | loggedOnSteam                  | Logged into steam as Revadike (76561198042965266)
 2019-04-04 | 00:20:26.176 | DEBUG | emit                           | Sending message: ClientChangeStatus
 2019-04-04 | 00:20:26.177 | DEBUG | emit                           | Sending message: ClientPICSChangesSinceRequest
 2019-04-04 | 00:20:26.178 | DEBUG | emit                           | Sending message: ClientRequestItemAnnouncements
 2019-04-04 | 00:20:26.179 | DEBUG | emit                           | Sending message: ClientRequestCommentNotifications 2019-04-04 | 00:20:26.180 | DEBUG | emit                           | Sending message: ClientChatRequestOfflineMessageCount
 2019-04-04 | 00:20:26.183 | DEBUG | emit                           | Handled message: ClientAccountInfo
 2019-04-04 | 00:20:26.184 | DEBUG | emit                           | Handled message: ClientAccountInfo
 2019-04-04 | 00:20:26.184 | DEBUG | emit                           | Handled message: ClientEmailAddrInfo
 2019-04-04 | 00:20:26.197 | DEBUG | emit                           | Handled message: ClientFriendsList
 2019-04-04 | 00:20:26.346 | DEBUG | emit                           | Sending message: ClientRequestFriendData
 2019-04-04 | 00:20:26.350 | DEBUG | emit                           | Handled message: ClientPlayerNicknameList
 2019-04-04 | 00:20:26.355 | DEBUG | emit                           | Unhandled message: ClientServersAvailable
 2019-04-04 | 00:20:26.411 | DEBUG | emit                           | Handled message: ClientFriendsGroupsList
 2019-04-04 | 00:20:26.421 | DEBUG | emit                           | Got incomplete message; expecting 234768 more bytes
 2019-04-04 | 00:20:26.422 | DEBUG | emit                           | Got incomplete message; expecting 234768 more bytes
 2019-04-04 | 00:20:26.423 | DEBUG | emit                           | Got incomplete message; expecting 234768 more bytes
 2019-04-04 | 00:20:26.423 | DEBUG | emit                           | Got incomplete message; expecting 234768 more bytes
```
Seems here it only needed 1 retry

**Comment by DoctorMcKay** [comment]
This sounds *very* similar to [this forum thread](https://dev.doctormckay.com/topic/2042-constantly-prompted-with-steam-guard-app-code/#entry6727).

TL;DR, add `"protocol": SteamUser.EConnectionProtocol.TCP` to your constructor options.

**Comment by Revadike** [comment]
That does seem to help. Will try to debug more.

**Comment by Revadike** [comment]
After several days of operation, I can confirm this solves it.
